### PR TITLE
Switch row layout edit modal #1774

### LINKED ISSUE #1774: Switch edit layout rows for easier access
**Is your feature request related to a problem? Please describe.**
It's very annoying when ypu want to quickly add sonething and have to find a tiny 'enter button' on a mobile keyboard - just to find the big orange 'Make goal' button.

**Describe the solution you'd like**
Switch layout rows so 'Make goal' button is not hidden by keyboard and accessible for quick add.

**Describe alternatives you've considered**
Add a button on the title edit bar. This would reduce the space available for the title.

**Additional context**
I know clicking on the empty space above the popup also works - but it's not intuitive - and not obvious for a first user.

### PR DESCRIPTION
**Pull request by harith-hacky03**
Switch edit layout rows for easier access #1774 is fixed Successfully

Resolves #1774

### CODE REVIEW

**Review by tijlleenders — CHANGES_REQUESTED** [review]
Thanks for the effort @harith-hacky03 but I don't see any layout change...

This is purely shifting things around in the layout so it should just be a switcharound.

I didn't expect any new imports either. Why do you need that?

**Comment by harith-hacky03** [comment]
@tijlleenders Now the keyboard is not hiding the create goal button since I moved the layout to the top in the mobile device. Is it okay ?

**Comment by tijlleenders** [comment]
@harith-hacky03 Yes, that's OK I guess. This was not documented as an alternative fix. The downside of this is that we lose the context of the parent goal.  

I guess we can evaluate your option for a while.

**Could you also switch the rows too?**  

We'll then merge it and use it for a while.
We can always revert one and/or the other if it reduces the overall experience.

**Comment by harith-hacky03** [comment]
@tijlleenders Sure ! I am ready to do it. Can you please provide some details with image with rows wants to be changed . Thank you

**Comment by tijlleenders** [comment]
Sure @harith-hacky03 I guess the image in the issue is not clear enough.  

Here's the desired state:
image

**Comment by harith-hacky03** [comment]
@tijlleenders Yeah! Thank you

**Comment by harith-hacky03** [comment]
@tijlleenders  Yeah i changed the rows . Hope it helps. If any changes is needed ping me.

**Review by tijlleenders — APPROVED** [review]
Excellent @harith-hacky03 !

@Tushar-4781 OK on code too?

**Comment by harith-hacky03** [comment]
@tijlleenders Is it good now ?

**Comment by tijlleenders** [comment]
@harith-hacky03 Yeah it's great.  

Just asking @Tushar-4781 to validate the code before we merge. I mostly look at the UI/UX/feature side of things.

**Comment by harith-hacky03** [comment]
@tijlleenders  Thank you.

File: package.json
Comment (Tushar-4781) [inline package.json:88]: please discard this as its not in the scope of this PR

File: src/common/ZModal.tsx
Comment (Tushar-4781) [inline src/common/ZModal.tsx:22]: umm, i don't see any change . any specific reason to add this check ? why no just keep it centered by default ?

File: src/common/ZModal.tsx
Comment (Tushar-4781) [inline src/common/ZModal.tsx:7]: can you remove this please.

File: package-lock.json
Comment (Tushar-4781) [inline package-lock.json:1]: please discard this as well

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]
hi Harith, thanks for your contributions. can you please check the following comments and then we will be gtg

File: package-lock.json
Comment (harith-hacky03) [reply package-lock.json:1]: Is it okay now?

**Comment by tijlleenders** [comment]
@harith-hacky03 

CI is failing with message `npm ERR! `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.`

**Comment by harith-hacky03** [comment]
@tijlleenders Is it okay now?

**Comment by tijlleenders** [comment]
@harith-hacky03 Yup, all good:
image

@Tushar-4781 will merge if all code issues are confirmed resolved.

Thanks.

File: package-lock.json
Comment (Tushar-4781) [reply package-lock.json:1]: @harith-hacky03 please discard the package and lock files from this PR and then we are gtg

Comment (harith-hacky03) [reply package-lock.json:1]: I wanna remove the types/crypto file from the package.json?

Comment (harith-hacky03) [reply package-lock.json:1]: @Tushar-4781 Or wanna remove the package.json file . please provide some details in brief

**Comment by tijlleenders** [comment]
@harith-hacky03 The only thing blocking this PR is the modified package files.  
If you look at the '[files changed by this pr](https://raw.githubusercontent.com/tijlleenders/ZinZen/main/package-lock.json)' - you can see the package files are not the same.  

Since you're merging from a fork I can't fix this for you.  
Just replace the contents of the `package.json` and `package-lock.json` with the code from main branch - and there will be no diffs: 
- https://github.com/tijlleenders/ZinZen/blob/main/package-lock.json
- https://raw.githubusercontent.com/tijlleenders/ZinZen/main/package.json

**Comment by harith-hacky03** [comment]
@tijlleenders  Is it okay now?!

**Review by tijlleenders — APPROVED** [review]
Great!

Thanks a bunch @harith-hacky03 ;)

**Comment by harith-hacky03** [comment]
@tijlleenders  Thank you for the good guidance. Can you merge it now without any difficulties?!

**Pull request merged**
